ddtrace/settings/dynamic_instrumentation.py

```diff
@@ -82,5 +89,30 @@ class DynamicInstrumentationConfig(En):
         help="Interval in seconds for periodically emitting probe diagnostic messages",
     )
 
+    redacted_identifiers = En.v(
+        set,
+        "redacted_identifiers",
+        map=normalize_ident,
+        default=set(),
+        help_type="List",
+        help="List of identifiers/object attributes/dict keys to redact from dynamic logs and snapshots",
+    )
+
+    redacted_types = En.v(
+        set,
+        "redacted_types",
+        map=str.strip,
+        default=set(),
+        help_type="List",
+        help="List of object types to redact from dynamic logs and snapshots",
+    )
+
+    redacted_types_re = En.d(
+        t.Optional[re.Pattern],
+        lambda c: re.compile(f"(?:{'|'.join((_.replace('.', '[.]').replace('*', '.*') for _ in c.redacted_types))})")
```

Comment:
I love this optimization. very cool.

I think we should sanitized "\[](){}?" etc. from the user input. maybe just allow [\w.*] and throw away all other values.

Reply:
Ah I'm not expecting the user to be allowed to use anything other than `*` for now, as per the RFC.